scripts/server.js

```diff
@@ -0,0 +1,143 @@
+const fs = require('fs');
+const path = require('path');
+
+const chalk = require('chalk');
+const express = require('express');
+const proxy = require('http-proxy-middleware');
+const cookiesMiddleware = require('universal-cookie-express');
+
+// Load environment configuration.
+process.env.NODE_ENV = process.env.NODE_ENV || 'development';
+require('react-scripts/config/env');
+
+const ROOT_PATH = path.join(__dirname, '..');
+const PROXY_PORT = 3100;
+
+// Create an express server.
+const app = express();
+
+// Express configuration.
+app.set('port', process.env.PORT || 3000);
+app.disable('x-powered-by');
+app.use(cookiesMiddleware());
+
+// This function injects the authentication token into the HTML sent to the
+// browser (`index.html`). The token is stored into a HttpOnly cookie.
+const injectAuthenticationToken = (req, html) => {
+  // Try to retrieve the authentication cookie, which contains the token.
+  const authToken = req.universalCookies.get(
+    process.env.REACT_APP_AUTHENTICATION_COOKIE,
+  );
+
+  // Replace the placeholder in the `index.html` file with the token.
+  return html.replace(
+    `"${process.env.REACT_APP_AUTH_TOKEN_PLACEHOLDER}"`,
+    JSON.stringify(authToken || '').replace(/</g, '\\u003c'),
+  );
+};
+
+// We use a proxy to forward API requests to REACT_APP_API_HOST (i.e. the
+// AMO/addons-server API). This is useful for local development.
+if (process.env.REACT_APP_REQUIRE_PROXY === 'true') {
```

Comment:
We can't really rely on `NODE_ENV` because `NODE_ENV` is not our configuration environment, it is eith `development` or `production`, and it means more compiled or not compiled code than anything else :/

Adding a warning makes sense though.